lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -29,102 +29,81 @@
 
   var mailboxList = <Mailbox>[].obs;
   var mailboxHasRoleList = <Mailbox>[].obs;
-  var mailboxMyFolderList = <MailboxFolder>[].obs;
+  var mailboxMyFolderList = <Mailbox>[].obs;
 
-  MailboxController(this._getAllMailboxInteractor);
+  MailboxController(this._getAllMailboxInteractor, this._acceptDataInterceptors);
 
   @override
   void onReady() {
     super.onReady();
+    _acceptDataInterceptors.changeAcceptData(Constant.acceptJMap);
     getAllMailboxAction();
   }
   
   void getAllMailboxAction() async {
+    final JMapAccountId.AccountId accountId = JMapAccountId.AccountId(JMapId.Id('3ce33c876a726662c627746eb9537a1d13c2338193ef27bd051a3ce5c0fe5b12'));
+
     mailboxState.value = MailboxState.LOADING;
-    await _getAllMailboxInteractor.execute()
+    await _getAllMailboxInteractor.execute(accountId)
       .then((response) => response.fold(
         (failure) => mailboxState.value = MailboxState.FAILURE,
         (success) {
           mailboxState.value = MailboxState.SUCCESS;
           mailboxList.value = success is GetAllMailboxViewState ? success.mailboxList : [];
           _setListMailboxHasRole(mailboxList);
-          _setListMailboxMyFolder(mailboxList);
+          _setListMailboxOfMyFolder(mailboxList);
         }));
   }
 
   void _setListMailboxHasRole(List<Mailbox> mailboxesList) {
     mailboxHasRoleList.value = mailboxesList
       .where((mailbox) => mailbox.role != MailboxRole.none)
       .toList();
+
+    final mailboxInBox = mailboxHasRoleList.where((mailbox) => mailbox.role == MailboxRole.inbox).toList();
+    if (mailboxInBox.isNotEmpty) {
+      selectMailbox(mailboxInBox.first);
+    }
   }
 
-  void _setListMailboxMyFolder(List<Mailbox> mailboxesList) {
-    final mapsMyFolder = HashMap<JMapMailbox.MailboxId, List<MailboxFolder>>();
-    final listMailboxFolder = mailboxesList
+  void _setListMailboxOfMyFolder(List<Mailbox> mailboxesList) {
+    final listMailboxOfMyFolder = mailboxesList
       .where((mailbox) => mailbox.role == MailboxRole.none)
       .toList();
 
-    JMapMailbox.MailboxId? parentID;
-    var listChildMailbox = <MailboxFolder>[];
-
-    for (int i = 0; i < listMailboxFolder.length; i++) {
-      final mailbox = listMailboxFolder[i];
-
-      if (mailbox.isRootFolder()) {
-        parentID = mailbox.parentId;
-        listChildMailbox = <MailboxFolder>[];
-      } else {
-        listChildMailbox.add(mailbox.toMailboxFolder([]));
-      }
-
-      mapsMyFolder.addIf(() => parentID != null, parentID!, listChildMailbox);
-    }
-
-    final listFolderRoot = listMailboxFolder.where((folder) => folder.isRootFolder()).toList();
-
-    mailboxMyFolderList.value = listFolderRoot
-      .map((mailbox) => mailbox.toMailboxFolder(mailbox.isRootFolder()
-        ? mapsMyFolder[mailbox.parentId!] ?? []
-        : []))
+    mailboxMyFolderList.value = listMailboxOfMyFolder
+      .map((mailbox) => mailbox.toMailboxParent(mailbox.getNameMailboxFolderHasParentId(mailboxesList)))
       .toList();
```

Comment:
How does this behave for someone having 1000+ mailboxes :trollface: ?

(Please take 1-2h to check that, this is quitte common, and known to have had caused issues, eg for the INBOX mailbox tree)

Reply:
The problem you said I was thinking about. But because the data returned from the server is organized in a sequential structure. So I use the algorithm to find the parent node of the child node here sequentially. I think it's a temporary solution. The user will wait when the data is finished loading. I designed the loading button so that the user understands the data being updated.
If you have a more optimal solution for this search. Please suggest me.

Reply:
I think your algorithm is in `O(n2)` to build the tree.

While a good implementation can achieve `O(ln(n))` (we do achieve it on James server side).

I would encorage at least to create a ticket to refine the algorithm.

Reply:
Please give me the source code link where you got O(ln(n)) search algorithm

Reply:
> Please give me the source code link where you got O(ln(n)) search algorithm

It will likely not translate as the use case is not exactly the same (say if a mailbox have children or not)

 - We do create a hashmap sized for the mailbox count
 - We add the mailboxes in the hash map on a first pass
 - We then call the hashmap for each mailbox to know if it has children or not
 
It does the job well for us.

Reply:
I think you are wrong here. As you say, you run the loop n here is the algorithm time is O(n2).

Reply:
nln(n) on the first pass to build the map

\+

n to iterate that map

All good ;-)

Reply:
I need the source code you handle this. I want to verify it again.

Reply:
https://github.com/apache/james-project/blob/e1e854ea7292a8cdc6e9c1dd6d8fb4cd261e6678/mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java#L653

Until line 782. Good luck.